framework/src/org/checkerframework/common/value/Range.java

```diff
@@ -0,0 +1,433 @@
+package org.checkerframework.common.value;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * The Range Class with mathematics operations. Models the range indicated by the @IntRange
+ * annotation.
+ *
+ * <p>Note that this class is not responsible for detecting incorrect parameters, e.g. the value of
+ * "from" could be greater than the value of "to". This incorrectness would be eventually caught by
+ * {@link
+ * org.checkerframework.common.value.ValueAnnotatedTypeFactory#createIntRangeAnnotation(Range)} when
+ * creating an annotation from range, which would then be replaced with @UnknownVal.
+ *
+ * @author JasonMrX
+ */
+public class Range {
+
+    /** The value 'from' */
+    public final long from;
+
+    /** The value 'to' */
+    public final long to;
+
+    /**
+     * Constructs a range with its bounds specified by two parameters, "from" and "to".
+     *
+     * <p>Note that it is possible to construct a range with incorrect parameters, e.g. the value of
+     * "from" could be greater than the value to "to". This incorrectness would be caught by {@link
+     * org.checkerframework.common.value.ValueAnnotatedTypeFactory#createIntRangeAnnotation(Range)}
```

Comment:
I think so too, but I just can't think of the answers for the following questions:
1. What would be a good choice for the special `NOTHING` constant? If we forbid illegal representation, then every combination of "from" and "to" is a candidate for `NOTHING` as long as "from" is greater than "to".
2. If the corresponding annotation is `@Bottom` if the range is illegal, what would be the result of a operation that involves `@Bottom` and other annotations? For example, what is `@Bottom` + `@IntRange`? `@Bottom` as well or do we just crash the checker at this point?

Any suggestions?

Besides, it's "by design" (by myself..) that empty intersection corresponds to top. The reason is that I don't want the analysis to be "terminated", since an expression that involves `@Bottom` would forever evaluate to `@Bottom` IMO.